src/Worker/Core/Shims/TaskEntityShim.cs

```diff
@@ -130,29 +131,30 @@ public void Commit()
         public void Rollback()
         {
             this.CurrentState = this.checkpointValue;
+            this.cachedValue = null;
```

Comment:
I agree that clearing the cache explicitly seems like a good idea here.

(since the code may have modified the object without calling `SetState` and then the comparison in the setter does not catch that it was modified).